parser-typechecker/src/Unison/Codebase/SqliteCodebase/Migrations/MigrateSchema5To6.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedLists #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Unison.Codebase.SqliteCodebase.Migrations.MigrateSchema5To6 (migrateSchema5To6) where
+
+import Control.Monad.Except
+import Control.Monad.State
+import Data.String.Here.Uninterpolated (here)
+import qualified U.Codebase.Sqlite.DbId as DB
+import qualified U.Codebase.Sqlite.DbId as Db
+import qualified U.Codebase.Sqlite.Operations as Ops
+import qualified U.Codebase.Sqlite.Queries as Q
+import qualified U.Codebase.Sync as Sync
+import qualified Unison.Debug as Debug
+import qualified Unison.Sqlite as Sqlite
+
+-- | Adds a table for tracking namespace statistics
+-- Adds stats for all existing namespaces, even though missing stats are computed on-demand if missing.
+migrateSchema5To6 :: Sqlite.Transaction ()
+migrateSchema5To6 = do
+  Q.expectSchemaVersion 5
+  Q.addNamespaceStatsTables
+  addStatsToAllNamespaces
+  Q.setSchemaVersion 6
+
+addStatsToAllNamespaces :: Sqlite.Transaction ()
```

Comment:
I decided to go ahead and just compute stats for all namespaces as part of the migration, but if in the future we're missing stats from a sync or something those will be computed when needed.

Reply:
👍